### PR TITLE
{tools}[foss/2015b] python-openstackclient-3.12.0 (REVIEW)

### PR DESCRIPTION
**Pull request by nathanhaigh**

### CODE REVIEW

File: easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+version = '3.12.0'
+
+homepage = "https://pypi.python.org/pypi/python-openstackclient/"
+description = """OpenStack Client"""
+
+toolchain = {'name': 'foss', 'version': '2015b'}
+
+python = 'Python'
+pythonversion = '2.7.11'
+py_short_ver = '.'.join(pythonversion.split('.')[0:2])
```
Comment (boegel) [inline easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb]: You're not actually using this `py_short_ver` variable at all, please include `Python` directly as dependency & use the `%(pyver)s` and ``%(pyshortver)s` templates:

```python
version = '3.12.0'
versionsuffix = '-Python-%(pyver)s'
...
dependencies = [
    ('Python', '2.7.11'),
    ...
]

File: easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+versionsuffix = "-%s-%s" % (python, pythonversion)
+
+exts_defaultclass = 'PythonPackage'
```
Comment (boegel) [inline easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb]: please move this closer to `exts_list`

File: easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+dependencies = [
+    (python, pythonversion),
+    ('libffi', '3.2.1'),    # for pyOpenSSL
+]
+
+#osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
```
Comment (boegel) [inline easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb]: Why is this commented out?

File: easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+exts_list = [
+#    ('', '', {
+#        'source_urls': ['https://pypi.python.org/packages/source/'],
+#    }),
```
Comment (boegel) [inline easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb]: please remove?

File: easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    'files': ['bin/openstack'],
+    'dirs': ['lib/python%(pyshortver)s/site-packages'],
+}
+
+sanity_check_commands = [('openstack', '--help')]
+
+modextrapaths = {'PYTHONPATH': ['lib/python%(pyshortver)s/site-packages']}
+
+moduleclass = 'tools'
+
```
Comment (boegel) [inline easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb]: please drop trailing empty line

**Comment by boegel** [comment]
(close/open to trigger Travis to run the tests...)

File: easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb (hunk 3 above)
Comment (nathanhaigh) [reply easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb]: I used one of the `Python` easyconfigs as a template. Given that `pyOpenSSL` is a dependency, I wasn't sure if this was required or not so left it in. Can you comment @boegel ?

Comment (boegel) [reply easybuild/easyconfigs/p/python-openstack/python-openstackclient-3.12.0-foss-2015b-Python-2.7.11.eb]: `pyOpenSSL` is just a wrapper around `OpenSSL`, so OpenSSL is indeed required, and we should prefer not installing it as a dependency to ensure version updates are picked up from the OS...

**Comment by boegel** [comment]
Hmm, for some reason Travis isn't picking up on this...

**Comment by migueldiascosta** [comment]
@nathanhaigh, this PR is being closed for the following reason(s): no activity for > 1 year.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**